compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1315,16 +1315,16 @@ impl<'a, 'tcx> InferCtxtPrivExt<'a, 'tcx> for InferCtxt<'a, 'tcx> {
                         | ObligationCauseCode::OpaqueType
                 );
                 // FIXME(associated_const_equality): Handle Consts here
-                let data_ty = data.term.ty().unwrap();
+                let data_ty = data.term;
                 if let Err(error) = self.at(&obligation.cause, obligation.param_env).eq_exp(
                     is_normalized_ty_expected,
                     normalized_ty,
                     data_ty,
                 ) {
                     values = Some(infer::ValuePairs::Types(ExpectedFound::new(
```

Comment:
There is `ValuePairs::Const`, which should somewhat resolve your `FIXME` above. There is still the case of trying to use an assoc const in a projection where a type is expected. I don't know if this is reachable, so for now you can just `span_bug!` (and maybe add a test?)

Reply:
Will add a test